### PR TITLE
[scanning] implement digit container for serial number input

### PR DESCRIPTION
**Pull request by allisonhongberkeley**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
Changed serial input text container to single digit, OTP-style input. 

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy!"
IMG_1653
IMG_1654

## How to review
[//]: # "Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?"

Open up the scanning page and try to add vouchers manually. Ensure that both valid + invalid vouchers both have their expected behavior. 

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

### Online sources
[//]: # "Optional - copy links to any tutorials or documentation that was useful to you when working on this PR"

https://www.npmjs.com/package/react-native-otp-textinput

### Related PRs
[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

[//]: # "This tags the project leader as a default. Feel free to change, or add on anyone who you should be in on the conversation."
🥦 CC: @sauhardjain

### CODE REVIEW

File: src/screens/scanning/ManualVoucherScreen.tsx [hunk 1]
```diff
@@ -116,13 +116,21 @@ export default function ManualVoucherScreen({
...
+              textInputStyle={{
+                borderWidth: 1,
+                borderRadius: 2,
+                width: 30,
+              }}
               isValid={!showInvalidError}
-              keyboardType="number-pad"
-            />
+            >
+              {}
```
Comment (selene-huang) [inline src/screens/scanning/ManualVoucherScreen.tsx]: not sure if there's a special reason we need this on IOS, but on android it doesn't seem to be any different than just having \<OTPTextInput options \/>?

**Review by selene-huang — CHANGES_REQUESTED** [review]
lgtm, just have a quick question

File: src/screens/scanning/ManualVoucherScreen.tsx (hunk 1 above)
Comment (sauhardjain) [reply src/screens/scanning/ManualVoucherScreen.tsx]: good catch, should work across devices — it's just a react thing

File: package.json [hunk 2]
```diff
@@ -28,13 +28,17 @@
     "react-native-dotenv": "^3.3.1",
     "react-native-dropdown-picker": "^5.4.3",
     "react-native-element-dropdown": "^2.5.0",
+    "react-native-otp-form": "^1.1.2",
+    "react-native-otp-inputs": "^7.4.0",
```
Comment (sauhardjain) [inline package.json]: Let's uninstall these unused libraries

File: package.json [hunk 3]
```diff
@@ -28,13 +28,17 @@
...
+    "react-native-otp-textinput": "^1.0.0",
     "react-native-paper": "^4.12.5",
     "react-native-safe-area-context": "4.4.1",
     "react-native-screens": "~3.18.0",
     "react-native-toast-message": "^2.1.5",
     "react-native-vector-icons": "^9.2.0",
     "react-native-web": "~0.18.7",
     "react-navigation": "^4.4.4",
+    "react-otp-input": "^3.0.0",
```
Comment (sauhardjain) [inline package.json:39]: Uninstall this too

**Review by sauhardjain — CHANGES_REQUESTED** [review]
Looks great, just one fix required — currently the input fields don't clear out after successfully adding a serial number. Can you try to fix this?

File: package.json (hunk 2 above)
Comment (allisonhongberkeley) [reply package.json]: done

**Review by selene-huang — APPROVED** [review]
lgtm!

**Review by sauhardjain — APPROVED** [review]
GREAT WORKKK looks good to merge!
